recipe_scrapers/bestrecipes.py

```diff
@@ -48,4 +50,10 @@ def prep_time(self):
         return self.schema.prep_time()
 
     def language(self):
+        msg = (
```

Comment:
The way you set the message here differs a little from the way we did it in my PR -- I think I put it at the top as a variable so that it could be reused in other functions within the module (even though it isn't). Since these messages seem to be used somewhat often between modules, I was thinking we could define another module with a few custom warning classes or messages. This would make the warning messages more standardized. We could also use them in testing https://docs.python.org/3/library/warnings.html#testing-warnings, which might make them a little less noisy during testing.

Reply:
@rmdluo I took some inspiration from your branch and went a step further, [in this PR](https://github.com/hhursev/recipe-scrapers/pull/1098) I have put it as a function in `_utils`. I'm currently investigating creating a custom warning category so that we could turn off those warnings during testing.

Reply:
We can use catch_warnings to test that keys in the test json with a null value should raise a warning. This silences the warnings from the user side, but also tests to make sure that the warning is there. I have something implemented for this, but I'm not sure if we should make a separate pull request for this?

Reply:
basic idea is this:

```
if expect[key]:
    self.assertEqual(
        expect[key],
        scraper_func(),
        msg=f"The actual value for .{key}() did not match the expected value.",
    )
else:
    with warnings.catch_warnings(record=True) as w:
        warnings.simplefilter("always")
        scraper_func()
        assert len(w) >= 1
        assert issubclass(w[-1].category, UserWarning) # can be adjusted to accomodate other warning types
```

Reply:
Very much agreed with the idea of detecting this automatically using `catch_warnings` in the unit tests :+1: 

...however, value-is-`null` isn't exactly the condition I had in mind to detect; it's value-is-always-the-same (it's kinda a warning that the scraper isn't doing anything smart for that particular field, or really even returning anything from the page itself - it's simply returning a fixed value that is _hopefully_ correct -- and worth double-checking -- for all recipes read by a scraper).

Reply:
I guess for silencing the warnings, we could move catch_warnings to above the assert equal. For checking warning messages with non-null values, you might have to add more information to the test jsons to denote that the value stays the same.

Reply:
> I guess for silencing the warnings, we could move catch_warnings to above the assert equal.

:+1: yep, ok - that sounds like a good idea to me

> For checking warning messages with non-null values, you might have to add more information to the test jsons to denote that the value stays the same.

I'd prefer to encourage developers to provide this information in the scraper implementations.  We'll probably forget to annotate some scrapers, but _maybe_ with some clever linting/tooling we could, in future, detect methods that always return a fixed/literal value.